xlators/storage/posix/src/posix-helpers.c

```diff
@@ -742,13 +794,20 @@ posix_istat(xlator_t *this, inode_t *inode, uuid_t gfid, const char *basename,
         goto out;
     }
 
+again:
     ret = sys_lstat(real_path, &lstatbuf);
 
     if (ret != 0) {
         if (ret == -1) {
-            if (errno != ENOENT && errno != ELOOP)
+            if (errno != ENOENT && errno != ELOOP) {
                 gf_msg(this->name, GF_LOG_WARNING, errno, P_MSG_LSTAT_FAILED,
                        "lstat failed on %s", real_path);
+            } else {
```

Comment:
This should only be done if `errno == ENOENT`. For `errno == ELOOP` we should fail (though I'm not sure if we can get this error in any case).

Reply:
done